rewrite-java/src/main/java/org/openrewrite/java/internal/parser/TypeTable.java

```diff
@@ -235,24 +252,17 @@ private void writeClassesDir(@Nullable GroupArtifactVersion gav, Map<ClassDefini
                         classDef.getSuperinterfaceSignatures()
                 );
 
-                Set<ClassDefinition> innerClasses = new HashSet<>();
-                for (ClassDefinition innerClassDef : membersByClassName.keySet()) {
```

Comment:
Note: This here was the problematic part. I could also only have made changes to this loop, but this quadratic overhead didn't feel right.